lsp/nls/src/linearization.rs

```diff
@@ -64,6 +78,40 @@ impl BuildingExt for Linearization<Building<BuildingResource>> {
         };
     }
 
+    fn register_fields(
+        &mut self,
+        record_fields: &HashMap<Ident, RichTerm>,
+        record: usize,
+        scope: Vec<ScopeId>,
+        env: &mut Environment,
+    ) {
+        for (ident, value) in record_fields.into_iter() {
+            let id = self.id_gen().take();
```

Comment:
The semantics of the idGen are indeed a bit weird i admit.
`take()` was chosen as it takes the current id, after which it becomes "inaccessible " i.e. the next id is prepared. I thought about `next()` but that sounds like it would give you the increased value. In that sense `take()` resembles a `i++` while `next()` (at least in my mind) is a `++i`

Reply:
IMO the increased value is not relevant to the API of `IdGen`. How does `IdGen` generate unique consecutive IDs is implementation detail. The thing is, as a consumer of `IdGen`, what this function does is to provide you with the next consecutive & unique id available, which is why I tend to think `next` would make more sense. Even more because `take` is used a lot for something quite different in Rust (taking ownership of an inner value, replacing it with a default one).

Reply:
Hm I think it is somewhat important to know that a call to `take` means the **next** call to `id` will give a new/different id `take`. I see though how "take" is an overloaded term in rust, but so is "next" (especially thinking about the semantics). how about "advance" or "get_and_advance"?

Reply:
Oh, because there's also a method to get the current `id`. My bad then. I personally like `advance` better, yes (a bit like `remove`, it does something and return the state before we did the thing). But you know, names and color. Do as you think is best.

Reply:
I documented the IdGen methods a bit and call tehm `get`/`get_and_advance` that should be clearer now